### PR TITLE
[mercedesme] Remove the access token when the thing is removed

### PR DESCRIPTION
**Pull request by lolodomo**
Related to #14818

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

File: bundles/org.openhab.binding.mercedesme/src/main/java/org/openhab/binding/mercedesme/internal/handler/AccountHandler.java [hunk 1]
```diff
@@ -123,11 +123,22 @@ private String configValid() {
...
+            server = Optional.empty();
             Utils.removePort(config.get().callbackPort);
         }
     }
 
+    @Override
+    public void handleRemoval() {
+        if (!server.isEmpty()) {
+            server.get().deleteServiceAndAccessToken();
+        }
```
Comment (jlaur) [inline bundles/org.openhab.binding.mercedesme/src/main/java/org/openhab/binding/mercedesme/internal/handler/AccountHandler.java:138]:
```suggestion
        server.ifPresent(s -> s.deleteServiceAndAccessToken());
```

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks! Only two minor suggestions. I'll let @weymann have a look also.

File: bundles/org.openhab.binding.mercedesme/src/main/java/org/openhab/binding/mercedesme/internal/server/CallbackServer.java [hunk 2]
```diff
@@ -64,6 +67,16 @@ public CallbackServer(AccessTokenRefreshListener l, HttpClient hc, OAuthFactory
         INVALID_ACCESS_TOKEN.setAccessToken(Constants.EMPTY);
     }
 
+    public void dispose() {
+        oAuthFactory.ungetOAuthService(config.clientId);
+        AUTH_MAP.remove(Integer.valueOf(config.callbackPort));
+        SERVER_MAP.remove(Integer.valueOf(config.callbackPort));
+    }
+
+    public void deleteServiceAndAccessToken() {
```
[the remark this answers is not in the document]
Comment (lolodomo) [reply bundles/org.openhab.binding.mercedesme/src/main/java/org/openhab/binding/mercedesme/internal/server/CallbackServer.java:76]: Renamed

File: bundles/org.openhab.binding.mercedesme/src/main/java/org/openhab/binding/mercedesme/internal/handler/AccountHandler.java (hunk 1 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.mercedesme/src/main/java/org/openhab/binding/mercedesme/internal/handler/AccountHandler.java:138]: Changed

**Review by jlaur — APPROVED** [review]

**Review by weymann — APPROVED** [review]
lgtm

**Pull request merged**